Commit dei vari metodi

LoreTozz opened the pull request:
Ho implementato tutti i metodi e tutti i TODO richiesti. Ho aggiunto commenti per facilitare la lettura del codice.

LoreTozz committed:
Prima implementazione

Creato il metodo costruttore e il metodo printBoard

LoreTozz committed:
Costruttore funzionante

Ho corretto il costruttore precedentemente inviato

LoreTozz committed:
Implementato metodo lifeOnPlanet()

LoreTozz committed:
Implementato metodo oneTurn con le "vecchie regole"

prima versione del programma

LoreTozz committed:
aggiunta di commenti al metodo oneTurn

LoreTozz committed:
piccole correzioni di logica in class Planet

Djack1010 reviewed (commented):
Il codice mi sembra ok, passo a scaricarlo, compilarlo e testarlo.

Djack1010 reviewed (changes requested):
La soluzione non è corretta, prova a semplificare gli IF di oneTurn() e riscrivere le condizioni. Lascio alcuni esempi di input e successive iterazioni corrette che dovrebbe generare:
100101011->100010110->000001111->000001001->000000000
011010110->011101110->010111010->001101100->001111100->001000100->000000000

Dati gli stessi due esempi, la tua soluzione restituisce:
100101011->100000010->000000000
011010110->011111000->010100000->000000000

LoreTozz committed:
Game of life funziona

Ho modificato la logica e adesso funziona, le nostre soluzioni sono leggermente diverse perché ho inserito il fattore Random della nascita spontanea

Djack1010 commented on Planet.java:
```diff
@@ -42,73 +42,51 @@ public boolean lifeOnPlanet(){
+    		if(vicinoSx == 0 && board[i] == 1 && vicinoDx == 0) arraySupporto[i] = 0;
+    		if(vicinoSx == 0 && board[i] == 0 && vicinoDx == 0) {
+    				if(rand.nextInt(100) < 90) arraySupporto[i] = 0;
+    				else arraySupporto[i] = 1;
```
Aggiunta di randomicità nell'esercizio, passaggio non richiesto che risolve comunque il problema dei cicli infiniti. Consiglio altrimenti l'aggiunta di un limitatore al numero di turni dentro GameOfLife1D.java (nella condizione del while, ad esempio "board.getTurn() < 30")

Djack1010 commented:
Con le modifiche fatte alla parte di logica del metodo oneTurn(), la soluzione risulta adesso corretta :+1:
